tezos/new_context/src/hash/mod.rs

```diff
@@ -217,21 +169,39 @@ fn hash_long_inode(
 
             hasher.update(&[1u8]); // type tag
             leb128::write::unsigned(&mut hasher, *depth as u64)?;
-            leb128::write::unsigned(&mut hasher, *children as u64)?;
-            hasher.update(&[pointers.len() as u8]);
+            leb128::write::unsigned(&mut hasher, *nchildren as u64)?;
+            hasher.update(&[*npointers as u8]);
 
             // Inode pointer:
+
             //
             // |    1    |   32   |
             // +---------+--------+
             // |  index  |  hash  |
 
-            for (index, hash) in pointers {
-                hasher.update(&[*index]);
-                let hash = store
-                    .get_hash(*hash)?
-                    .ok_or_else(|| HashingError::HashIdNotFound { hash_id: *hash })?;
-                hasher.update(hash.as_ref());
+            for (index, pointer) in pointers.iter().enumerate() {
+                if let Some(pointer) = pointer.as_ref() {
```

Comment:
When is this `None` and what happens in that case?

Reply:
`Inode::Pointers` contains 32 pointers (`Option<PointerToInode>`), when one of them is `None`, it means that there is no entries on that index

Reply:
Ok, please add a small comment there to make it easier to understand what is going on.

Reply:
@tizoc Done